Foundation/include/Poco/ScopedLock.h

```diff
@@ -75,12 +75,12 @@ class ScopedLockWithUnlock
 	/// unlocking of the mutex.
 {
 public:
-	explicit ScopedLockWithUnlock(M& mutex): _pMutex(&mutex)
+	explicit ScopedLockWithUnlock(M& mutex): _pMutex(&mutex), _locked(true)
 	{
 		_pMutex->lock();
```

Comment:
Technically, the mutex is not locked until _pMutec->lock() completes. I propose setting _locked to true after _pMutex->lock() succeeds.